rest/rest-server/src/test/java/functionaltests/utils/RestFuncTUtils.java

```diff
@@ -42,17 +46,27 @@
 import org.ow2.proactive.resourcemanager.common.RMConstants;
 import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
 
+import com.jayway.awaitility.Duration;
+
 
 public class RestFuncTUtils {
 
     private RestFuncTUtils() {
     }
 
-    public static void destroy(Process process) throws Exception {
+    public static void destroy(Process process) throws InterruptedException {
+
         close(process.getOutputStream());
         close(process.getInputStream());
         close(process.getErrorStream());
-        process.destroyForcibly();
+
+        process.destroy();
+
+        if (!process.waitFor(1, TimeUnit.MINUTES)) {
+            process.destroyForcibly();
```

Comment:
I think this change is a good move. However one minute might be long. Did you notice a significant increase of the pull request time?

Reply:
process.waitFor(1, TimeUnit.MINUTES) waits for AT MOST one minute. If the process ends earlier, this instruction finishes and returns true. So, it has no impact on the execution time, unless the process takes more than one minute to finish.

Before, this instruction was 'process.waitFor()', which may wait endlessly.

Reply:
Ok, this is a very good improvement then!